### PR TITLE
kubeadm: Default to control plane v1.6.0-alpha.1 and using RBAC

### PR DESCRIPTION
**Pull request by luxas**
Also use constants for authz modes

**What this PR does / why we need it**:

Defaults to v1.6.0-alpha.1 (will be cut later today) because the certificates API group has been upgraded to beta, so `kubeadm join` at HEAD doesn't work on a `v1.5` cluster anyway.

By defaulting to v1.6.0-alpha.1, we can focus totally on v1.6 for kubeadm at HEAD, we don't support other versions in the upcoming v1.6 kubeadm release because of the alpha -> beta upgrades.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

@mikedanese @pires

### CODE REVIEW

File: cmd/kubeadm/app/cmd/defaults.go [hunk 1]
```diff
@@ -29,8 +29,7 @@ import (
 )
 
 var (
-	// Maximum version when using AllowAll as the default authz mode. Everything above this will use RBAC by default.
-	allowAllMaxVersion = semver.MustParse("1.6.0-alpha.0")
+	minK8sVersion = semver.MustParse("1.6.0-alpha.1")
```
Comment (pires) [inline cmd/kubeadm/app/cmd/defaults.go]: Isn't this another constant? `constants.DefaultKubernetesFallbackVersion` perhaps?

File: cmd/kubeadm/app/constants/constants.go [hunk 2]
```diff
@@ -28,4 +28,9 @@ const (
 	APIServerKubeletClientCertAndKeyBaseName = "apiserver-kubelet-client"
 	APIServerKubeletClientCertName           = "apiserver-kubelet-client.crt"
 	APIServerKubeletClientKeyName            = "apiserver-kubelet-client.key"
+
+	AuthzModeAlwaysAllow = "AlwaysAllow"
```
Comment (pires) [inline cmd/kubeadm/app/constants/constants.go]: If we're enforcing 1.6 and defaulting to RBAC, why is this needed then?

**Review by pires — APPROVED** [review]
Overall lgtm. Two nits though.

File: cmd/kubeadm/app/cmd/defaults.go (hunk 1 above)
Comment (luxas) [inline cmd/kubeadm/app/cmd/defaults.go]: It can't be a `const` since it's assigned by an external package `semver`.

And it's not the same as the fallback version either; in the future we might want to set the fallback version to be `v1.6.2` for instance, but the minimum required version would still be `v1.6.0` or the like

File: cmd/kubeadm/app/constants/constants.go (hunk 2 above)
Comment (luxas) [inline cmd/kubeadm/app/constants/constants.go]: You're right; it's not used anywhere, but I guess it might be there for completeness, since we were told to copy these constants from another package (which also have these constants). They were copied over here to kubeadm until a subpackage is made for the authorizer that includes only the constants.

**Comment by luxas** [comment]
@k8s-bot cvm gke e2e test this

**Comment by luxas** [comment]
@k8s-bot cvm gce e2e test this
@k8s-bot gci gce e2e test this
@k8s-bot gci gke e2e test this
@k8s-bot gce etcd3 e2e test this

File: cmd/kubeadm/app/cmd/defaults.go (hunk 1 above)
Comment (pires) [inline cmd/kubeadm/app/cmd/defaults.go]: I meant the string `"1.6.0-alpha.1"`.

File: cmd/kubeadm/app/constants/constants.go (hunk 2 above)
Comment (pires) [inline cmd/kubeadm/app/constants/constants.go]: Hmmmm OK.

File: cmd/kubeadm/app/cmd/defaults.go (hunk 1 above)
Comment (luxas) [inline cmd/kubeadm/app/cmd/defaults.go]: Ah, ok. Yeah, makes sense. I can do that in a follow up, I've already rebased upon this commit in my larger PR

File: cmd/kubeadm/app/constants/constants.go (hunk 2 above)
Comment (luxas) [inline cmd/kubeadm/app/constants/constants.go]: Sorry for that strange comment above; I'm a little bit tired :)
Basically we have to duplicate the constants unless we want the binary to grow twice the size or so because of all the other packages that would follow. This can be fixed in the future when the authorizer package is a bit more refactored

Comment (luxas) [inline cmd/kubeadm/app/constants/constants.go]: cc @liggitt @deads2k

**Comment by luxas** [comment]
@k8s-bot cvm gce e2e test this
@k8s-bot gci gce e2e test this
@k8s-bot gce etcd3 e2e test this

File: cmd/kubeadm/app/cmd/defaults.go (hunk 1 above)
Comment (mikedanese) [inline cmd/kubeadm/app/cmd/defaults.go]: Seems small enough to do now.

File: cmd/kubeadm/app/cmd/defaults.go [hunk 3]
```diff
@@ -53,15 +52,17 @@ func setInitDynamicDefaults(cfg *kubeadmapi.MasterConfiguration) error {
...
-	if k8sVersion.GT(allowAllMaxVersion) {
-		cfg.AuthorizationMode = "RBAC"
+	if err != nil {
+		return fmt.Errorf("couldn't parse kubernetes version: %v", err)
+	}
+	if k8sVersion.LT(minK8sVersion) {
+		return fmt.Errorf("this version of kubeadm only supports deploying clusters with the control plane version >= v1.6.0-alpha.1. Current version: %s", cfg.KubernetesVersion)
 	}
 
+	fmt.Println("[init] Using Kubernetes version:", ver)
```
Comment (mikedanese) [inline cmd/kubeadm/app/cmd/defaults.go]: nit please use printf

**Comment by mikedanese** [comment]
lgtm, please make the small fix, squash then feel free to self apply.

**Comment by luxas** [comment]
Self-applying; got to removing some other, now unnecessary code as well

**Comment by pires** [comment]
@k8s-bot unit test this
